services/leecher/leecher/leecher.py

```diff
@@ -24,18 +23,51 @@ def create_event_description(event):
 
 
 def callback(event):
-    if event.topic in IGNORE_TOPICS:
+    """Callback function for Aquarium event listener.
+    Checks the event data for item, gets corresponding project, then dispatches an event to Ayon.
+    """
+
+    # get aq item from event
+    event_data = event.to_dict()
+    item = event_data["data"].get("item")
+    item_key = item.get("_id")
+    if not all([item, item_key]):
+        log.error(f"Can't retrieve Aquarium item from event: {event_data}")
         return
+    aq_item = _AQS.aq.cast(item)
```

Comment:
I realize that [the class Event](https://github.com/fatfish-lab/aquarium-python-api/blob/c42cdcd00dac16bcdefd38d04eaf2cbe552e8375/aquarium/events.py#L227) in our documentation is missing.

Here is a suggestion to simplify the code and also handle all use case, since `event.data.item` is not mandatory.
```suggestion
    # get aq item from event
    context = event.get_context()
```

Reply:
can i assume that the item of interest that has been leeched from aquarium is always the very first?
so can i assume this:
`item = event.get_context()["path"][0]`

Reply:
addressed in https://github.com/ynput/ayon-aquarium/commit/f8301de725faf409bb99514434dc91a2de448dfa

Reply:
> can i assume that the item of interest that has been leeched from aquarium is always the very first? so can i assume this: `item = event.get_context()["path"][0]`

Yes you can assume this, it's correct. :)